sdk/src/azure/core/az_span.c

```diff
@@ -592,13 +607,22 @@ AZ_NODISCARD az_result _az_span_replace(
   // insert at the end case (no need to make left or right shift)
   if (start == current_size)
   {
+    if (url_encode)
+    {
+      return _az_span_url_encode(
+          az_span_slice_to_end(destination, start), replacement, out_url_encode_size);
+    }
     destination = az_span_copy(az_span_slice_to_end(destination, start), replacement);
     return AZ_OK;
   }
   // replace all content case (no need to make left or right shift, only copy)
   // TODO: Verify and fix this check, if needed.
   if (current_size == replaced_size)
   {
+    if (url_encode)
+    {
+      return _az_span_url_encode(destination, replacement, out_url_encode_size);
```

Comment:
I think this might be incorrect and will override the data in the destination that we don't want replaced.

What is the `current_size` parameter used for? It isn't documented in the header file.

**Let's say:**
- current size = 6
- start = 2
- end = 8

Then replaced_size = 6, i.e. `current_size == replaced_size`.

But then, when we encode the replacement and copy it into the destination, we could potentially write beyond the end of 8 because encoding could expand the data.

**Here's a more concrete example:**
buffer: "test.com/q1=+++++++++&q2=abc .... < empty space > ..."
**And we call replace with these values:**
- current size = 9
- start = 12
- end = 21

**We expect the `&q2...` to be shifted:**
buffer: "test.com/q1=%2B%2B%2B%2B%2B%2B%2B%2B%2B&q2=abc  .... < empty space > ..."

**Instead we will get something incorrect, like this, where `&q2...` got over-written:**
buffer: "test.com/q1=%2B%2B%2B%2B%2B%2B%2B%2B%2B  .... < empty space > ..."

It is quite possible that I am understanding the expected behavior and parameters incorrectly, and I am not sure what will actually happen exactly, but something to test and look into to verify we are getting the behavior we want.

Reply:
OK, so `current_size` is the entire length of the URL so start/end cannot be larger than it. **Never mind, you can ignore the above scenario.**